### PR TITLE
limit the requested reports to the last seven days by using the minCr…

### PR DESCRIPTION
**Pull request by MichaelRoytman**
…eatedDate query parameter on the reports API endpoint in registrar

### CODE REVIEW

File: src/common/utils.js [hunk 1]
```diff
@@ -79,3 +79,16 @@ export class AsyncActionType {
...
+  const date = new Date();
+  date.setUTCDate(date.getUTCDate()-days);
+  return date;
+}
+
+export function getISODateString(date) {
+  const isoDate = date.toISOString();
+  // we only want the date, not the time
+  const isoDateParts = isoDate.split('T')
+  return isoDateParts[0]
```
Comment (MichaelRoytman) [inline src/common/utils.js]: I had originally tried something along the lines of:

```
  const day = date.getUTCDate()
  // getUTCMonth is 0-indexed
  const month =  date.getUTCMonth() + 1;
  const year = date.getUTCFullYear();

  return `${year}-${month}-${day}`;
```
however, month is zero indexed in Javascript's Date class, which is annoying, and formatting the ISO string myself was more difficult than I expected (e.g. I had trouble getting a leading zero to show for dates 1-9). I have decided on this approach instead. `toISOString`is based on the standard ISO 8601 simplified format, so I feel comfortable splitting on the `T` character.

File: src/common/utils.js [hunk 2]
```diff
@@ -79,3 +79,16 @@ export class AsyncActionType {
     return `${this.topic}__${this.name}__RESET`;
   }
 }
+
+export function getTodayWithDaysOffset(days) {
+  const date = new Date();
+  date.setUTCDate(date.getUTCDate() - days);
```
Comment (schenedx) [inline src/common/utils.js]: Why do we need to convert UTCDate back and forth? Why can't we just do `offsetDay = date - days`?

File: src/common/utils.test.js [hunk 3]
```diff
@@ -115,4 +124,37 @@ describe('keepKeys', () => {
       expect(actionType.RESET).toBe('HOUSE_CATS__START_THE_RACE__RESET');
     });
   });
+
+  describe('getDateWithDaysOffset', () => {
```
Comment (kdmccormick) [inline src/common/utils.test.js]: I think what makes your `getDateWithDaysOffset` interesting and non-trivial is that it wraps to the previous month/year correctly, which isn't obvious from reading its definition. Consider adding a test case (or just changing your existing one) to demonstrate that `getTodayWithDaysOffset` works on, say, January 1st.

File: src/console/constants.js [hunk 4]
```diff
@@ -6,3 +6,10 @@ export const PERMISSIONS = { // eslint-disable-line import/prefer-default-export
   writeEnrollments: 'write_enrollments',
   readReports: 'read_reports',
 };
+
+// The number of days counting back from today's date to use for
+// limiting the reports requested by their creation date.
+// Note that this is inclusive of the current date;
+// therefore, a value of 6 represents 7 days of reports.
+// This value should be 0 or a positive integer.
+export const REPORT_DATE_OFFSET_DAYS = 6;
```
Comment (kdmccormick) [inline src/console/constants.js]: Thanks for the exposition on this :+1:

File: src/common/utils.test.js (hunk 3 above)
Comment (MichaelRoytman) [reply src/common/utils.test.js]: That's a good point. I'll add those those tests. Thanks.

**Pull request merged**